server/src/main/java/org/elasticsearch/action/admin/indices/shrink/ResizeRequest.java

```diff
@@ -20,114 +20,91 @@
 package org.elasticsearch.action.admin.indices.shrink;
 
 import java.io.IOException;
-import java.util.Objects;
+import java.util.ArrayList;
+import java.util.List;
 
-import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.Version;
 import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
-import org.elasticsearch.action.support.ActiveShardCount;
-import org.elasticsearch.action.support.IndicesOptions;
 import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.settings.Settings;
+
+import io.crate.metadata.IndexName;
+import io.crate.metadata.IndexParts;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RelationName;
 
 
 /**
- * Request class to shrink an index into a single shard
+ * Request resize of a table or partition
  */
-public class ResizeRequest extends AcknowledgedRequest<ResizeRequest> implements IndicesRequest {
+public class ResizeRequest extends AcknowledgedRequest<ResizeRequest> {
 
-    private CreateIndexRequest targetIndexRequest;
-    private String sourceIndex;
-    private ResizeType type = ResizeType.SHRINK;
-    private Boolean copySettings;
+    private final RelationName table;
+    private final List<String> partitionValues;
+    private final int newNumShards;
 
-    public ResizeRequest(String targetIndex, String sourceIndex) {
-        this.targetIndexRequest = new CreateIndexRequest(targetIndex, Settings.EMPTY);
-        this.sourceIndex = sourceIndex;
+    public ResizeRequest(RelationName table, List<String> partitionValues, int newNumShards) {
+        this.table = table;
+        this.partitionValues = partitionValues;
+        this.newNumShards = newNumShards;
     }
 
     public ResizeRequest(StreamInput in) throws IOException {
         super(in);
-        targetIndexRequest = new CreateIndexRequest(in);
-        sourceIndex = in.readString();
-        type = in.readEnum(ResizeType.class);
-        copySettings = in.readOptionalBoolean();
+        if (in.getVersion().onOrAfter(Version.V_5_10_0)) {
+            table = new RelationName(in);
+            int numValues = in.readVInt();
+            partitionValues = new ArrayList<>(numValues);
+            for (int i = 0; i < numValues; i++) {
+                partitionValues.add(in.readOptionalString());
+            }
+            newNumShards = in.readVInt();
+        } else {
+            CreateIndexRequest targetIndexRequest = new CreateIndexRequest(in);
+            String sourceIndex = in.readString();
+            in.readEnum(ResizeType.class);
```

Comment:
Could drop an Enum an inline dummy-read code but can do it later as well

Reply:
I'd rather keep it for readability - this isn't a hot loop. Resize operations are rare and resize operations during a feature upgrade even rarer.